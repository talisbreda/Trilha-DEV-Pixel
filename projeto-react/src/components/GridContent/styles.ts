import styled, { css } from 'styled-components';
import { Theme } from '../../styles/theme';

interface Props {
  theme: Theme;
}

export const Container = styled.div<Props>`
  ${({ theme }) => css`
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
  `}
`;

export const Html = styled.div<Props>`
  ${({ theme }) => css`
    margin: ${theme.spacings.huge} 0;
  `}
`;
