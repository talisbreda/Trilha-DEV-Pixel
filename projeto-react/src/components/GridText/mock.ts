export type GridTextElement = {
  title: string;
  description: string;
};

export type GridTextMock = {
  background: boolean;
  title: string;
  description: string;
  grid: Array<GridTextElement>;
};

export const mock: GridTextMock = {
  background: false,
  title: 'My grid',
  description:
    'Aliquam ac quam posuere, sagittis erat sit amet, eleifend felis. Sed ullamcorper ornare gravida.',
  grid: [
    {
      title: 'Teste 1',
      description: `Donec tincidunt tincidunt massa ac volutpat. Quisque vitae diam pretium
        porta purus nec, vulputate mauris. Vivamus vitae dapibus tortor. Ut eu gravida
        felis. In dictum dolor consequat, porta ante nec, viverra magna. Morbi nec erat eros.
        In vitae ipsum eget odio congue tristique non quis odio.'`,
    },
    {
      title: 'Teste 2',
      description: `Donec tincidunt tincidunt massa ac volutpat. Quisque vitae diam pretium
        porta purus nec, vulputate mauris. Vivamus vitae dapibus tortor. Ut eu gravida
        felis. In dictum dolor consequat, porta ante nec, viverra magna. Morbi nec erat eros.
        In vitae ipsum eget odio congue tristique non quis odio.'`,
    },
    {
      title: 'Teste 3',
      description: `Donec tincidunt tincidunt massa ac volutpat. Quisque vitae diam pretium
        porta purus nec, vulputate mauris. Vivamus vitae dapibus tortor. Ut eu gravida
        felis. In dictum dolor consequat, porta ante nec, viverra magna. Morbi nec erat eros.
        In vitae ipsum eget odio congue tristique non quis odio.'`,
    },
  ],
};
