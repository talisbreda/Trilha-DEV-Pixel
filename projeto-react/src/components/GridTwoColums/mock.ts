export default {
  title: 'Grid two columns',
  text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Autem itaque ullam, minus, non dolorem vitae atque doloremque assumenda voluptatem illum aliquid obcaecati voluptas omnis, maxime a culpa ea iusto ab!',
  imageSrc: 'assets/images/javascript.svg',
  background: false,
};
