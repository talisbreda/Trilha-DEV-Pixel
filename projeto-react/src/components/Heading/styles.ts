import styled, { css, FlattenSimpleInterpolation } from 'styled-components';
import { Theme } from '../../styles/theme';

interface Props {
  theme: Theme;
  darkColor: boolean;
  as: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
  size: 'huge' | 'large' | 'medium' | 'small';
  upperCase: boolean;
}

type sizes = {
  small: (theme: Theme) => FlattenSimpleInterpolation;
  medium: (theme: Theme) => FlattenSimpleInterpolation;
  large: (theme: Theme) => FlattenSimpleInterpolation;
  huge: (theme: Theme) => FlattenSimpleInterpolation;
};

const mediaFont = (theme: Theme) => css`
  @media ${theme.media.ltMedium} {
    font-size: ${theme.font.sizes.large}
  }
`;

const titleSize: sizes = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.small};
  `,
  medium: (theme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  large: (theme) => css`
    font-size: ${theme.font.sizes.large};
  `,
  huge: (theme) => css`
    font-size: ${theme.font.sizes.huge};
    ${mediaFont(theme)};
  `,
};

const titleCase = (upperCase: boolean) => css`
  text-transform: ${upperCase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1<Props>`
  ${({ theme, darkColor, size, upperCase }) => {
    return css`
        color: ${darkColor ? theme.colors.primaryColor : theme.colors.white};
        ${titleSize[size](theme)};
        ${titleCase(upperCase)};
    `;
  }}
`;
