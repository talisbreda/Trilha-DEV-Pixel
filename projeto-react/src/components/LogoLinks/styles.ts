import styled, { css } from 'styled-components';
import { Theme } from '../../styles/theme';

interface Props {
  theme: Theme;
  href: string;
}

export const Container = styled.a<Props>`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    text-decoration: none;
    color: inherit;

    > img {
      max-height: 3rem;
    }
  `}
`;
