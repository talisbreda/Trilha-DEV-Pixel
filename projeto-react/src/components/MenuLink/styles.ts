import styled, { css } from 'styled-components';
import { Theme } from '../../styles/theme';

interface Props {
  theme: Theme;
  href: string;
  target: string;
}

export const Container = styled.a<Props>`
  ${({ theme }) => css`
    display: block;
    text-decoration: none;
    font-size; ${theme.font.sizes.tiny};
    padding: ${theme.spacings.small};
    color: ${theme.colors.primaryColor};
    position: relative;

    &::after {
      content: '';
      position: absolute;
      bottom: 0.75rem;
      left: 50%;
      width: 0;
      height: 0.2rem;
      background: ${theme.colors.secondaryColor};
      transition: all 300ms ease-in-out;
    }
    &:hover::after {
      left: 25%;
      width: 50%;
    }
  `}
`;
