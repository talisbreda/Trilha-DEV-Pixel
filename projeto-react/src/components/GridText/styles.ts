import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';
import { Theme } from '../../styles/theme';
import { Title as HeadingContainer } from '../Heading/styles';

interface Props {
  theme: Theme;
}

export const Container = styled.div<Props>`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div<Props>`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large}
  `}
`;

export const GridElement = styled.div<Props>`
  ${({ theme }) => css`
    ${HeadingContainer} {
      position: relative;
      left: 5rem;
    }

    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: 7rem;
      top: -3rem;
      left: -5rem;
      transform: rotate(5deg);
    }
  `}
`;
