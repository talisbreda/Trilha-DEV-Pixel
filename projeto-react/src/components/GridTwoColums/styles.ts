import styled, { css } from 'styled-components';
import { Theme } from '../../styles/theme';
import { Title } from '../Heading/styles';

interface Props {
  theme: Theme;
}

export const Container = styled.div<Props>`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacings.large};

    @media ${theme.media.ltMedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }
    ${Title} {
      margin-bottom: ${theme.spacings.huge};
    }
  `}
`;

export const TextContainer = styled.div<Props>`
  ${({ theme }) => css`
    @media ${theme.media.ltMedium} {
      margin-bottom: ${theme.spacings.large};
    }
  `}
`;

export const ImageContainer = styled.div<Props>`
  ${() => css``}
`;

export const Image = styled.img<Props>`
  ${() => css``}
`;
