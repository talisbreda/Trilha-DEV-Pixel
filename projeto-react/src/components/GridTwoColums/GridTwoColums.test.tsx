import { renderTheme } from '../../styles/render-theme';
import { GridTwoColums } from '.';
import mock from './mock';

describe('<GridTwoColums />', () => {
  it('should render two column grid', () => {
    /* eslint-disable-next-line react/jsx-props-no-spreading */
    const { container } = renderTheme(<GridTwoColums {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
