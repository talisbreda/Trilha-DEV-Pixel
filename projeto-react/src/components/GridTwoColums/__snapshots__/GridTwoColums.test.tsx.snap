// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridTwoColums /> should render two column grid 1`] = `
<div>
  <div
    class="sc-dkrFOg cDrkPT"
  >
    <div
      class="sc-gswNZR irDqzv"
    >
      <div
        class="sc-eDvSVe cVzTBX"
      >
        <div
          class="sc-jSUZER dqokRe"
        >
          <h1
            class="sc-bcXHqe cLqjTh"
          >
            Grid two columns
          </h1>
          <div
            class="sc-hLBbgP eeZIjO"
          >
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Autem itaque ullam, minus, non dolorem vitae atque doloremque assumenda voluptatem illum aliquid obcaecati voluptas omnis, maxime a culpa ea iusto ab!
          </div>
        </div>
        <div
          class="sc-gKPRtg dQATfs"
        >
          <img
            alt="Grid two columns"
            class="sc-iBYQkv bWTXCf"
            src="assets/images/javascript.svg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
